src/Orleans/Runtime/RequestContext.cs

```diff
@@ -102,38 +115,30 @@ public static bool Remove(string key)
 
         public static void Import(Dictionary<string, object> contextData)
         {
+#if !NETSTANDARD
             if (PropagateActivityId)
             {
                 object activityIdObj = Guid.Empty;
-                if (contextData?.TryGetValue(E2_E_TRACING_ACTIVITY_ID_HEADER, out activityIdObj) != true)
+                if (contextData?.TryGetValue(E2_E_TRACING_ACTIVITY_ID_HEADER, out activityIdObj) == true)
                 {
-                    if (ActivityId.Value != Guid.Empty)
-                    {
-                        ActivityId.Value = Guid.Empty;
-                    }
-
-#if !NETSTANDARD
-                    object legacyActivityIdObj = null;
-                    if (contextData?.TryGetValue(E2_E_TRACING_LEGACY_ACTIVITY_ID_HEADER, out legacyActivityIdObj) == true)
-                    {
-                        Trace.CorrelationManager.ActivityId = (Guid) legacyActivityIdObj;
-                    }
-#endif
+                    Trace.CorrelationManager.ActivityId = (Guid)activityIdObj;
```

Comment:
Now we're going to store in ```ExecutionContex``` 2 guids - one through the ```CallContextData``` dictionary and second  - through ```Trace.CorrelationManager.ActivityId```. Shouldn't be an issue though.